guides/07_streaming/02_object-detection-from-webcam.md

```diff
@@ -1,93 +1,107 @@
+We'll download the YOLOv10 model from the Hugging Face hub and instantiate a custom inference class to use this model. 
+
+The implementation of the inference class isn't covered in this guide, but you can find the source code [here](https://huggingface.co/spaces/freddyaboulton/webrtc-yolov10n/blob/main/inference.py#L9) if you're interested. This implementation borrows heavily from this [github repository](https://github.com/ibaiGorordo/ONNX-YOLOv8-Object-Detection).
+
+**Tip:** We're using the `yolov10-n` variant because it has the lowest latency. See the [Performance](https://github.com/THU-MIG/yolov10?tab=readme-ov-file#performance) section of the README in the YOLOv10 GitHub repository.
 
 ```python
-def yolov10_inference(image, conf_threshold):
-    width, _ = image.size
-    results = model.predict(source=image, imgsz=width, conf=conf_threshold)
-    annotated_image = results[0].plot()
-    return annotated_image[:, :, ::-1]
+from huggingface_hub import hf_hub_download
+from inference import YOLOv10
+
+model_file = hf_hub_download(
+    repo_id="onnx-community/yolov10n", filename="onnx/model.onnx"
+)
+
+model = YOLOv10(model_file)
+
+def detection(image, conf_threshold=0.3):
+    image = cv2.resize(image, (model.input_width, model.input_height))
+    new_image = model.detect_objects(image, conf_threshold)
+    return new_image
 ```
 
-We will use the `plot` method to draw a bounding box around each detected object. YoloV10 asses images are in the BGR color format, so we will flip them to be in the expected RGB format of web browsers.
+Our inference function, `detection`, accepts a numpy array from the webcam and a desired confidence threshold. Object detection models like YOLO identify many objects and assign a confidence score to each. The lower the confidence, the higher the chance of a false positive. We'll let users adjust the confidence threshold.
+
+The function returns a numpy array corresponding to the same input image with all detected objects in bounding boxes.
 
 ## The Gradio Demo
 
-The Gradio demo will be pretty straightforward but we'll do a couple of things that are specific to streaming:
+The Gradio demo is straightforward, but we'll implement a few specific features:
 
-* The user's webcam will be both an input and an output. That way, the user will only see their stream with the detected objects.
-* We'll use the `time_limit` and `stream_every` parameters of the `stream` event. The `time_limit` parameter will mean that we'll process each user's stream for that amount of time. In a multi-user setting, such as on Spaces, this means that after this period of time, the next user in the queue will be able to use the demo. The `stream_every` function will control how frequently the webcam stream is sent to the server.
+1. Use the `WebRTC` custom component to ensure input and output are sent to/from the server with WebRTC. 
```

Comment:
Link to the webrtc custom component here?

Reply:
Done!